common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java

```diff
@@ -48,6 +49,15 @@ public long size() {
   }
 
   /**
+   * Reset the size of the memory block.
```

Comment:
I think it's a really strange and dangerous interface. Why do we have to set the size? Does the memory consumer complain when he gets a memory block larger than he requested?

Reply:
The size of memory block is not aligned size，resetting the size is just to keep the behavior consistent with before.